### PR TITLE
fix(security): mitigate Cross-Site Scripting attack (XSS).

### PR DESCRIPTION
**Pull request by bhaskarvilles**
Unsanitized input from an HTTP parameter flows into send, where it is used to render an HTML page returned to the user. This may result in a Cross-Site Scripting attack (XSS).

Signed-off-by: Bhaskar <dev@bhaskar.email>

### CODE REVIEW

**Review by izuru0 — COMMENTED** [review]
@bhaskarvilles User input `req.params.id` is fed to `escapeHtml` before `res.send` sees it. Are there cases that output from `escapeHtml` can still be harmful?

**Comment by bhaskarvilles** [comment]
> @bhaskarvilles User input `req.params.id` is fed to `escapeHtml` before `res.send` sees it. Are there cases that output from `escapeHtml` can still be harmful?

Yes, in some cases it happens, So we have to  Sanitize data input in an HTTP request before reflecting it back, ensuring all data is validated, filtered or escaped before echoing anything back to the user, such as the values of query parameters during searches. And also convert special characters into their respective HTML or URL encoded equivalents.

**Review by izuru0 — APPROVED** [review]
@bhaskarvilles ok, thanks!

LGTM

I've approved it, but do you mind if I asked for an example or url to docs including an example describing the risk you mentioned?

**Comment by bhaskarvilles** [comment]
> @bhaskarvilles ok, thanks!
> 
> LGTM
> 
> I've approved it, but do you mind if I asked for an example or url to docs including an example describing the risk you mentioned?

You can Find [here](https://cwe.mitre.org/data/definitions/79.html)

**Comment by bhaskarvilles** [comment]
@izuru0 
I think you haven't merged my pr and not referenced in any commit if merged ?

**Review by petermetz — APPROVED** [review]
@bhaskarvilles LGTM, thank you!

**Review by petermetz — CHANGES_REQUESTED** [review]
@bhaskarvilles Please squash the commits into a single one with an interactive rebase. Let me know if you need help and if you do make sure to invite me to your fork as a collaborator.

**Comment by bhaskarvilles** [comment]
> @bhaskarvilles Please squash the commits into a single one with an interactive rebase. Let me know if you need help and if you do make sure to invite me to your fork as a collaborator.

I've invited you as collaborators, please squash and commit changes.

**Review by petermetz — APPROVED** [review]
@bhaskarvilles LGTM, thank you!

FYI: I fixed the commits for you, please double check that you are also happy with them as-is before we merge.

**Comment by bhaskarvilles** [comment]
> @bhaskarvilles LGTM, thank you!
> 
> FYI: I fixed the commits for you, please double check that you are also happy with them as-is before we merge.

I'm happy to see my contributions publicly, always ❤️ Open-Source. Thank you for helping me in squashing @petermetz

**Pull request merged**
